models/User.php

```diff
@@ -69,16 +71,87 @@ public function behaviors()
     public function rules()
     {
         return [
-            ['status', 'default', 'value' => self::STATUS_ACTIVE],
-            ['status', 'in', 'range' => [self::STATUS_ACTIVE, self::STATUS_DELETED]],
             ['is_authenticated', 'boolean'],
-            ['name', 'string'],
             [['gender_id', 'sexuality_id', 'currency_id'], 'integer'],
+
+            ['status',
+                'default',
+                'value' => self::STATUS_ACTIVE],
+            ['status',
+                'in',
+                'range' => [self::STATUS_ACTIVE, self::STATUS_DELETED]],
+
             ['email', 'email'],
+            ['email', 'required'],
+            ['email',
+                'unique',
+                'message' => 'Email must be unique.'
+            ],
+
+            ['username', 'required'],
```

Comment:
@grandmotivator are `name` and `username` attributes required? In db they can be NULL at the moment. If they are required, I think we need to write a migration to make these fields NOT NULL